### PR TITLE
[16.01] Keep track of hidden datasets

### LINKED ISSUE #1534: When re-running a tool with default hidden input, changing parameters removes hidden inputs
Re-running a tool that uses a hidden dataset...

selection_548

Changing a parameter here removes the hidden inputs and defaults to another file.

selection_549

### PR DESCRIPTION
**Pull request by guerler**
Should fix #1534.

### CODE REVIEW

File: client/galaxy/scripts/mvc/ui/ui-select-default.js [hunk 1]
```diff
@@ -186,6 +187,13 @@ var View = Backbone.View.extend({
 
     /** Update all available options at once
     */
+    add: function( options, sorter ) {
+        _.each( this.model.get( 'options' ), function( v ) {
+            !_.findWhere( options, v ) && options.push( v );
+        });
```
Comment (carlfeberhard) [inline client/galaxy/scripts/mvc/ui/ui-select-default.js:193]: If `options` and `model.get( 'options' )` are both arrays, can't you use `_.union` here? http://underscorejs.org/#union

**Comment by carlfeberhard** [comment]
I understand you're trying to get a fix in, so +1 if it does what you need.

Otherwise, I'm kinda +/-0. I'm not sure why you need a generic model in there to maintain the options array if you're not responding to events with it.

**Comment by guerler** [comment]
@carlfeberhard Thx man. You are right. I have added the model preemptively as preparation for the next iteration which will contain proper listeners i.e. be analogous to the recent ui-misc modifications. Other than that it seems that _.union does not work with dictionaries?! since an equal operator/callback can not be specified.

**Comment by carlfeberhard** [comment]
@guerler Yeah - you're absolutely right - union won't work if `options` are both lists of dictionaries.

**Pull request merged**